packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -137,31 +140,65 @@ export class WorkspaceCommandContribution implements CommandContribution {
                 );
             })
         }));
-        registry.registerCommand(WorkspaceCommands.FILE_DELETE, this.newFileHandler({
-            execute: async uri => {
+        registry.registerCommand(WorkspaceCommands.FILE_DELETE, this.newMultiUriAwareCommandHandler({
+            execute: async uris => {
+                const msg = (() => {
+                    if (uris.length === 1) {
+                        return `Do you really want to delete ${uris[0].path.base}?`;
+                    }
+                    if (uris.length > 10) {
+                        return `Do you really want to delete all the ${uris.length} selected files?`;
+                    }
+                    const messageContainer = document.createElement('div');
+                    messageContainer.textContent = 'Do you really want to delete the following files?';
+                    const list = document.createElement('ul');
+                    list.style.listStyleType = 'none';
+                    for (const uri of uris) {
+                        const listItem = document.createElement('li');
+                        listItem.textContent = uri.path.base;
+                        list.appendChild(listItem);
+                    }
+                    messageContainer.appendChild(list);
+                    return messageContainer;
+                })();
                 const dialog = new ConfirmDialog({
-                    title: 'Delete File',
-                    msg: `Do you really want to delete '${uri.path.base}'?`
+                    title: `Delete File${uris.length === 1 ? '' : 's'}`,
+                    msg,
                 });
                 if (await dialog.open()) {
-                    await this.fileSystem.delete(uri.toString());
+                    // Make sure we delete the longest paths first, they might be nested. Longer paths come first.
```

Comment:
I've noticed that VS Code tried to find the most common folders to delete. It could be more efficient to filter out children of such folders.

Reply:
OK.

Reply:
@akosyakov, what if the FS would take care of it? I could change the API from `delete(uri: string, options?: { moveToTrash?: boolean }): Promise<void>;` to `
delete(uri: string | string[], options?: { moveToTrash?: boolean }): Promise<void>;`.

The only problem I can see with the API change, what should we do if one of the file deletions fails. I can throw a specific error showing all the URIs, the URIs which have been deleted, and the URIs which not. Something like this.

This would go into a separate GitHub task.

Reply:
ok, let's work on it separately, throwing not so nice, since all errors end up in the log

Reply:
https://github.com/theia-ide/theia/issues/1476